src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -15,19 +15,16 @@ bool HasByReferenceArgs(const TaskSpecification &spec) {
 }
 
 CoreWorkerDirectActorTaskSubmitter::CoreWorkerDirectActorTaskSubmitter(
-    boost::asio::io_service &io_service, gcs::RedisGcsClient &gcs_client,
+    boost::asio::io_service &io_service,
     std::unique_ptr<CoreWorkerStoreProvider> store_provider)
     : io_service_(io_service),
-      gcs_client_(gcs_client),
       client_call_manager_(io_service),
       store_provider_(std::move(store_provider)) {}
 
-Status CoreWorkerDirectActorTaskSubmitter::SubmitTask(
-    const TaskSpecification &task_spec) {
+void CoreWorkerDirectActorTaskSubmitter::SubmitTask(const TaskSpecification &task_spec) {
   RAY_LOG(DEBUG) << "Submitting task " << task_spec.TaskId();
-  if (HasByReferenceArgs(task_spec)) {
-    return Status::Invalid("Direct actor call only supports by-value arguments");
-  }
+  RAY_CHECK(!HasByReferenceArgs(task_spec))
```

Comment:
Why not return Status instead of CHECK crashing?

Reply:
Because this shouldn't ever happen except when debugging. This method as is can only fail if the task spec has by-reference args, and I don't think it makes sense to have it return a `Status` just because of that possibility. I think the right thing to do long-term is to only enable `RAY_CHECK`s in development mode.

Reply:
Won't it RAY_CHECK if you pass an actor handle out of band?

Reply:
You mean like pass an actor ID as an argument? I'm not really sure what would happen in that case. I can just have this return a status again.